src/Compilation.zig

```diff
@@ -851,14 +851,60 @@ pub fn create(gpa: Allocator, options: InitOptions) !*Compilation {
             break :blk false;
         };
 
-        const sysroot = blk: {
-            if (options.sysroot) |sysroot| {
-                break :blk sysroot;
-            } else if (options.native_darwin_sdk) |sdk| {
-                break :blk sdk.path;
-            } else {
-                break :blk null;
+        var clang_argv = std.ArrayList([]const u8).init(arena);
+        var lib_dirs = std.ArrayList([]const u8).init(arena);
+        var framework_dirs = std.ArrayList([]const u8).init(arena);
+        var rpath_list = std.ArrayList([]const u8).init(arena);
```

Comment:
Can you try adding a commit that adds the needed state here (except clang_argv) to `LibCDirs` and moving the corresponding logic to `detectLibCIncludeDirs` or `getZigShippedLibCIncludeDirsDarwin`?

I think that might clean this up even more.